LenovoLegionToolkit.Lib/Listeners/HDRListener.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Threading.Tasks;
+using LenovoLegionToolkit.Lib.Extensions;
+using LenovoLegionToolkit.Lib.System;
+using LenovoLegionToolkit.Lib.Utils;
+using Microsoft.Win32;
+
+namespace LenovoLegionToolkit.Lib.Listeners;
+
+public class HDRListener : IListener<HDRListener.ChangedEventArgs>
+{
+    public class ChangedEventArgs(HDRState message) : EventArgs
+    {
+        public HDRState Message { get; } = message;
+    }
+
+    private bool _started;
+
+    public event EventHandler<ChangedEventArgs>? Changed;
+    
+    public bool IsHDROn { get; private set; }
+
+    public bool IsOK { get; private set; }
+
+    public Task StartAsync()
+    {
+        if (_started)
+            return Task.CompletedTask;
+
+        var display = InternalDisplay.Get();
+        if (display is null)
+        {
+            if (Log.Instance.IsTraceEnabled)
+                Log.Instance.Trace($"Built in display not found");
+
+            IsOK = false;
+        }
+        else
+        {
+            IsHDROn = display.GetAdvancedColorInfo().AdvancedColorEnabled;
+            IsOK = true;
+        }
+        SystemEvents.DisplaySettingsChanged += SystemEvents_DisplaySettingsChanged;
+        _started = true;
+
+        return Task.CompletedTask;
+    }
+
+    public Task StopAsync()
+    {
+        SystemEvents.DisplaySettingsChanged -= SystemEvents_DisplaySettingsChanged;
+        _started = false;
+
+        return Task.CompletedTask;
+    }
+    
+    private void SystemEvents_DisplaySettingsChanged(object? sender, EventArgs e)
+    {
+        if (Log.Instance.IsTraceEnabled)
+            Log.Instance.Trace($"Event received.");
+
+        Task.Delay(100).ContinueWith(_ =>
```

Comment:
In this architecture both `HDRListener` and `DisplayConfigurationListener` are watching `SystemEvents.DisplaySettingsChanged` signal. The thing (which caused the unreliability before) is, `HDRListener` needs `InternalDisplay` class to get HDR state, but `DisplayConfigurationListener` calls `InternalDisplay` class to refresh, and they run in parallel. Normally, `HDRListener` gains the lock in `InternalDisplay` first and get what it needs, and then `InternalDisplay` will be set to refresh-needed, which causes the display api `HDRListener` got becomes null. Therefore I added this delay here, in order to let `InternalDisplay` set its state first.